rules/framework/vfs_overlay.bzl

```diff
@@ -11,11 +11,33 @@ VFSOverlayInfo = provider(
     },
 )
 
+# Computes the "back" segment for a path length
+def _make_relative_prefix(length):
+    dots = "../"
+    prefix = ""
+    for i in range(0, length + 1):
+        prefix += dots
+    return prefix
+
+# The LLVM `VirtualFileSystem` object can serialize paths relative to the
+# absolute path of the overlay. This requires the paths are relative to the
+# overlay. While deriving the in-memory tree roots, it pre-pends the prefix of
+# the `vfsoverlay` path to each of the entries.
+def _get_external_contents(prefix, path_str):
+    return prefix + path_str
+
 # Make roots for a given framework. For now this is done in starlark for speed
 # and incrementality. For imported frameworks, there is additional search paths
 # enabled
 def _make_root(ctx, framework_name, root_dir, extra_search_paths, module_map, hdrs, private_hdrs, has_swift):
     extra_roots = []
+
+    # Compute with the build file path, assuming the build file path is adjacent
```

Comment:
This is a fair assumption. There is one flaw in this plan: the memoization of this vfs will no longer be valid if the entire VFS must be relative